src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -1229,6 +1239,11 @@ private void setOption_completionSource(IntPtr optionsPtr, IntPtr completionSour
                 var increaseTo = newPartitionsForTopic.IncreaseTo;
                 var assignments = newPartitionsForTopic.ReplicaAssignments;
 
+                if (newPartitionsForTopic.Topic == null)
+                {
+                    throw new NullReferenceException("Cannot assign partition to null topic.");
```

Comment:
Again, `ArgumentException` and maybe for the text "Cannot add partitions to a null topic." (word assign isn't appropriate).

this also leaks memory. i think we need a finally block like this:

```
finally {
    foreach (var newPartitionPtr in newPartitionsPtrs)
    {
        if (newPartitionPtr != IntPtr.Zero) {
            Librdkafka.NewPartitions_destroy(newPartitionPtr);
        }
    }
}
```

the current code is not really ideal because it too can leak memory, but this happens only in the event of an OOM error, so everything is screwed anyway.

what's different here is we leak memory on invalid args.

Reply:
I've implemented this pattern on the methods touched by this pr. 

Glancing through a couple other methods (like DescribeConfigs), it looks like we might need to do a review for other areas needing this kind of disposal strategy as well.

Reply:
I've implemented this strategy in most of the methods touched by this pull request.  

I saw one or two other methods, like DescribeConfigs, that I believe could also use this disposal strategy.